disperser/cmd/encoder/flags/flags.go

```diff
@@ -113,6 +113,18 @@ var (
 		Value:    string(encoding.GnarkBackend),
 		EnvVar:   common.PrefixEnvVar(envVarPrefix, "BACKEND"),
 	}
+	GPUConcurrentFrameGenerationDangerousFlag = cli.IntFlag{
+		Name: common.PrefixFlag(FlagPrefix, "gpu-concurrent-frame-generation-dangerous"),
+		Usage: "Weight of the GPU semaphore. Increase this value to allow more concurrent GPU tasks. " +
+			"This is only used by V2 when GPU is enabled (Backend=icicle and GPUEnable=true). " +
+			"Chunk generation (encoding/v2/rs) and multiproofs generation (encoding/v2/kzg/prover) " +
+			"each have their own separate semaphore which is weighted using this value. " +
+			"WARNING: setting this value too high may lead to out-of-memory errors on the GPU. " +
+			"If this ever happens, the GPU device needs to be rebooted as it can be left in a bad state.",
+		Required: false,
+		Value:    1,
```

Comment:
What happens if icicle is enabled but GPUConcurrentFrameGenerationDangerousFlag is set to 0?
The benchmark sets this to 20. What is a reasonably safe range for a single L40/A10?

Reply:
1. you get a panic. Added config verification to encoder's constructor: [e77b650](https://github.com/Layr-Labs/eigenda/pull/2249/commits/e77b65096ee9b4d95a932b7f4dfbfaa646e23585)

Reply:
as for your second question... we'll really have to experiment here, I'm not sure. For a L40/A10 we have 24GiB of GPU RAM. After setup we send afair <2GiB of srs points and ntt domain points, so we have 22GiB left to play with. FFT uses 2*blob_size of RAM, so at most 256MiB (16MiB * 8codingRatio * 2). MSM I honestly am not sure, would need to benchmark and play around with it more. Ive added https://dev.ingonyama.com/api/cpp/msm#memory-usage-estimation to the docs. Could do some more research if you're interested.